httpclient/httpclient.go

```diff
@@ -26,3 +31,138 @@ func CloneHTTPHeader(in http.Header) http.Header {
 	}
 	return out
 }
+
+// New wraps delegate transports with logging, rate limiting, retryable, request id
+// and returns an error if any occurs.
+func New(cfg *Config) (*http.Client, error) {
```

Comment:
Why can’t we simply call `NewWithOpts(cfg, Opts{})` within the `New` constructor?